OPHD/XmlSerializer.cpp

```diff
@@ -0,0 +1,24 @@
+#include "XmlSerializer.h"
+#include <NAS2D/Utility.h>
+#include <NAS2D/Filesystem.h>
+
+using namespace NAS2D;
+
+Xml::XmlDocument openXmlFile(std::string filename, std::string rootElementName)
+{
+	Xml::XmlDocument xmlDocument;
+	xmlDocument.parse(Utility<Filesystem>::get().open(filename).raw_bytes());
+
+	if (xmlDocument.error())
+	{
+		throw std::runtime_error(filename + " has malformed XML: Row: " + std::to_string(xmlDocument.errorRow()) +
+			" Column: " + std::to_string(xmlDocument.errorCol()) + " : " + xmlDocument.errorDesc());
+	}
+
+	const auto* xmlRootElement = xmlDocument.firstChildElement(rootElementName);
+	if (!xmlRootElement) {
+		throw std::runtime_error(filename + " does not contain required root tag of <" + rootElementName + ">");
+	}
+
+	return xmlDocument;
```

Comment:
I had assumed the xmlDocument needed to be passed out due to lifetime, but I didn't actually test. My original thought was what you were thinking about just passing the root element, but I didn't try it due to the lifetime consideration. Maybe we can test to see if it matters or put the xmlDocument into a unique_ptr to see if that would preserve it even if only the xmlRootElement pointer is passed out?

Reply:
I looked a bit at the XML code. It seems lifetime is tied to the `XmlDocument`, so if it's not being returned, or otherwise preserved, the root node would go out of scope upon return, and you'd be left with a dangling pointer.

I had a local branch in NAS2D where I was trying to deal with this sort of thing. I think I attempted to pass in a lambda for the processing code, which was called with the `rootElement`, rather than try to return it.

I think maybe ignore my original comment on this for now. This might be more relevant for consideration in a future PR.